src/adera/SysExhaustPlume.cpp

```diff
@@ -39,52 +41,135 @@
 
 using namespace adera::active;
 
-using osp::AssetImporter;
-
-using osp::DependRes;
-using osp::Package;
-
 using osp::active::ActiveScene;
 using osp::active::ActiveEnt;
 
+/**
+ * Attach a visual exhaust plume effect to MachineRocket
+ *
+ * Searches the hierarchy under the specified MachineRocket entity and
+ * attaches an ACompExhaustPlume to the rocket's plume node. A graphical
+ * exhaust plume effect will be attached to the node by SysExhaustPlume
+ * when it processes the component.
+ *
+ * @param rScene [ref] Scene containing the following entities
+ * @param part [in] Entity containing a plume in its descendents
+ * @param mach [in] Entity containing MachineRocket
+ */
+void attach_plume_effect(ActiveScene &rScene, ActiveEnt part, ActiveEnt mach)
+{
+    using osp::active::ACompName;
+    using osp::active::EHierarchyTraverseStatus;
+    using osp::active::SysHierarchy;
 
-using osp::active::ACompMesh;
-using osp::active::ACompShader;
-using osp::active::ACompTransparent;
-using osp::active::ACompVisible;
+    ActiveEnt plumeNode = entt::null;
 
-void SysExhaustPlume::initialize_plume(ActiveScene& rScene, ActiveEnt node)
-{
-    using Magnum::GL::Mesh;
-    using Magnum::GL::Texture2D;
-    using Magnum::Trade::ImageData2D;
-    using namespace Magnum::Math::Literals;
-    using adera::shader::PlumeShader;
+    auto findPlumeHandle = [&rScene, &plumeNode](ActiveEnt ent)
+    {
+        static constexpr std::string_view namePrefix = "fx_plume_";
+        auto const* name = rScene.reg_try_get<ACompName>(ent);
 
-    auto const& plume = rScene.reg_get<ACompExhaustPlume>(node);
+        if (nullptr == name)
+        {
+            return EHierarchyTraverseStatus::Continue;
+        }
+
+        if (0 == name->m_name.compare(0, namePrefix.size(), namePrefix))
+        {
+            plumeNode = ent;
+            return EHierarchyTraverseStatus::Stop;  // terminate search
+        }
+        return EHierarchyTraverseStatus::Continue;
+    };
+
+    SysHierarchy::traverse(rScene, part, findPlumeHandle);
 
-    if (rScene.get_registry().all_of<ACompShader>(node))
+    if (plumeNode == entt::null)
     {
-        return;  // plume already initialized
+        SPDLOG_LOGGER_ERROR(rScene.get_application().get_logger(),
+                          "ERROR: could not find plume anchor for MachineRocket {}", part);
+        return;
     }
 
-    DependRes<PlumeEffectData> plumeEffect = plume.m_effect;
+    SPDLOG_LOGGER_INFO(rScene.get_application().get_logger(), "MachineRocket {}\'s associated plume: {}",
```

Comment:
the ' needs escaping? That's really odd...?

Reply:
Unrelated to the ', i feel like we eventually want to have per-component logging, not application wide logging. But that doesn't need anything to happen here.,

Reply:
Systems are simple static functions and loggers are objects, which makes it 'difficult' to store the loggers. I'm thinking it's better for the loggers to be passed as arguments eventually. Or heck, maybe even a globally accessible per-thread 'currently selected logger.'

Reply:
that's fair too. could have a `thread_local std::reference_wrapper<Logger>` for the currently active logger (reference to indicate "not null").

How would we swap the logger though, if the different systems are being executed by a thread pool?

Logger can't be stored in entt? Certainly we could store std::reference_wrapper<Logger> in entt ?